tests/integration/test_remote_write_grafana_agent.py

```diff
@@ -36,14 +31,16 @@ async def test_remote_write_with_grafana_agent(ops_test, prometheus_charm):
         ),
     )
 
-    await ops_test.model.wait_for_idle(apps=apps, status="active")
-    assert initial_workload_is_ready(ops_test, apps)
+    await ops_test.model.wait_for_idle(apps=apps, status="active", wait_for_units=1)
     assert check_prometheus_is_ready(ops_test, prometheus_name, 0)
 
     await ops_test.model.add_relation(prometheus_name, agent_name)
-    await ops_test.model.wait_for_idle(apps=apps, status="active", idle_period=60)
 
-    await has_metric(
+    # A considerable idle_period is needed to guarantee metrics show up in prometheus
+    # (60 sec was not enough).
```

Comment:
A "Release to edge" workflow failed because of this:
https://github.com/canonical/prometheus-k8s-operator/runs/6611217860?check_suite_focus=true#step:4:1018